DataConnectors/BETTERMTD.json

```diff
@@ -0,0 +1,126 @@
+        {
+            "metricName": "Total incidents received",
+            "legend": "BetterMTDIncidentLog_CL",
+            "baseQuery": "BetterMTDIncidentLog_CL"
+        },
+        {
+            "metricName": "Total devcie enrolled",
+            "legend": "BetterMTDDeviceLog_CL",
+            "baseQuery": "BetterMTDDeviceLog_CL"
+        },
+        {
+            "metricName": "Total netflow traffic recieved",
+            "legend": "BetterMTDNetflowLog_CL",
+            "baseQuery": "BetterMTDNetflowLog_CL"
+        },
+        {
+            "metricName": "Total device applications",
+            "legend": "BetterMTDAppLog_CL",
+            "baseQuery": "BetterMTDAppLog_CL"
+        }
+    ],
+    "sampleQueries": [
+        {
+            "description": "All threats in the past 24 hour",
+            "query": "BetterMTDIncidentLog_CL\n            | where TimeGenerated > ago(24h)\n            | sort by TimeGenerated\n            | limit 100"
+        },
+        {
+            "description": "Enrolled Devices in the past 24 hour",
+            "query": "BetterMTDDeviceLog_CL\n            | where TimeGenerated > ago(24h)\n            | sort by TimeGenerated\n            | limit 100"
+        },
+        {
+            "description": "Installed applications in the last 24 hour",
+            "query": "BetterMTDAppLog_CL\n            | where TimeGenerated > ago(24h)  and  AppStatus_s  == \"installed\" \n            | sort by TimeGenerated            \n| limit 100"
+        },
+        {
+            "description": "Blocked Network traffics in the last 24 hour",
+            "query": "BetterMTDNetflowLog_CL\n            | where TimeGenerated > ago(24h)  and  Status_s == \"blocked\"\n            | sort by TimeGenerated\n            | limit 100"
+        }
+    ],
+    "dataTypes": [
+        {
+            "name": "BetterMTDIncidentLog_CL",
+            "lastDataReceivedQuery": "BetterMTDIncidentLog_CL\n            | summarize Time = max(TimeGenerated)\n            | where isnotempty(Time)"
+        },
+        {
+            "name": "BetterMTDDeviceLog_CL",
+            "lastDataReceivedQuery": "BetterMTDDeviceLog_CL\n            | summarize Time = max(TimeGenerated)\n            | where isnotempty(Time)"
+        },
+        {
+            "name": "BetterMTDAppLog_CL",
+            "lastDataReceivedQuery": "BetterMTDAppLog_CL\n            | summarize Time = max(TimeGenerated)\n            | where isnotempty(Time)"
+        },
+        {
+            "name": "BetterMTDNetflowLog_CL",
+            "lastDataReceivedQuery": "BetterMTDNetflowLog_CL\n            | summarize Time = max(TimeGenerated)\n            | where isnotempty(Time)"
+        }
+    ],
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "BetterMTDIncidentLog_CL\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
```

Comment:
The data connector will show as connected only if the specific table has logs, else even if other tables have logs and this doesn’t, (can happen if a certain type of logs are not there) it’ll show as disconnected. To fix this, multiple data types needs to be included as comma separated values in the connectivity query (it resolves as OR internally). We have a similar pattern for Carbon Black data connector (pasted below for quick reference).
           "value": [
	                "CarbonBlackEvents_CL\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)",
	                "CarbonBlackNotifications_CL\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)",
	                "CarbonBlackAuditLogs_CL\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
	            ]

Reply:
aymenim, thank you for the quick update. All other changes in this PR looks good.

Reply:
Thank you